src/middleware/router.js

```diff
@@ -553,11 +610,37 @@ function sendHttpRequest_OLD (ctx, route, options) {
       // }
 
       const bufs = []
-      routeRes.on('data', chunk => bufs.push(chunk))
+
+      if(!bucket) {
+        bucket = getGridFSBucket()
+      }
+
+      const uploadStream = bucket.openUploadStream()
+
+      uploadStream
+        .on('error', (err) => {
+          logger.error('Storing of response in gridfs failed, error: ' + JSON.stringify(err))
+        })
+        .on('finish', (file) => {
+          logger.info(`Response body with body id: ${file._id} stored`)
+
+          // Update HIM transaction with bodyId
+          ctx.response.bodyId = file._id
+        })
+
+      routeRes.on('data', chunk => {
+        if (!response.startTimestamp) {
+          response.startTimestamp = new Date()
+        }
+        uploadStream.write(chunk)
+        bufs.push(chunk)
+      })
 
       // See https://www.exratione.com/2014/07/nodejs-handling-uncertain-http-response-compression/
       routeRes.on('end', () => {
         response.timestamp = new Date()
+        response.endTimestamp = new Date()
```

Comment:
This property doesnt seem to be used anywhere within the codebase?

Reply:
Isn't we are going to have a start timestamp (when the stream starts) and an end timestamp (when the stream ends)?

Reply:
Sure, but when doing a search for that property it only appears where its defined, so its currently not sued anywhere. I also think the naming of the property elsewhere has been set to something like `timestampEnd` instead of `endTimestamp`